### PR TITLE
Merged commits management

### PR DESCRIPTION
**Pull request by csansoon**
- Added `mergedAt` property to commits
- Created `ProjectProvider` and `CommitProvider` components
- Modified `getDocumentsAtCommit` and other methods to new schema.

Note: Many functions require both `commitUuid` and `projectId`, since the `commitUuid` can be 'HEAD', and we will require the project to know which commit does that represent.

### CODE REVIEW

File: apps/web/src/app/(private)/layout.tsx [hunk 1]
```diff
@@ -14,12 +13,5 @@ export default async function PrivateLayout({
...
-  return (
-    <main className='flex flex-row w-full'>
-      <div className='w-[280px]'>
-        <Sidebar />
-      </div>
-      <div className='flex-1'>{children}</div>
-    </main>
-  )
+  return children
 }
```
Comment (csansoon) [inline apps/web/src/app/(private)/layout.tsx:17]: Sidebar has been moved to `/projects/[projectId]`

File: packages/web-ui/src/providers/CommitProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+interface CommitContextType {
+  commitUuid: string | typeof HEAD_COMMIT
+  isDraft: boolean
+}
+
+const CommitContext = createContext<CommitContextType>({
+  commitUuid: HEAD_COMMIT,
+  isDraft: false,
+})
```
Comment (csansoon) [inline packages/web-ui/src/providers/CommitProvider.tsx:15]: `useCurrentCommit` will return both the `commitUuid` and an `isDraft` property, which can be used to allow or forbid modifications depending on the commit.

File: packages/core/src/services/documentVersions/create.ts [hunk 3]
```diff
@@ -36,32 +41,37 @@ function createDocument({
...
+  const commitResult = await findCommit({ commitUuid, projectId })
+  const commit = commitResult.unwrap()
 
-      commit = resultCommit.value
-    }
+  if (commit.mergedAt !== null) {
+    return Result.error(
+      new ForbiddenError('Cannot create a document version in a merged commit'),
+    )
+  }
```
Comment (csansoon) [inline packages/core/src/services/documentVersions/create.ts:67]: Added this restriction on `createDocumentVersion`.

**Review by andresgutgon — APPROVED** [review]

File: packages/core/src/services/documentVersions/create.ts (hunk 3 above)
Comment (geclos) [reply packages/core/src/services/documentVersions/create.ts:67]: hmm would be cool to have model level validations like activerecord, @andresgutgon do u know if there's something for this in drizzle land?

Comment (andresgutgon) [reply packages/core/src/services/documentVersions/create.ts:67]: I saw this kind of repository pattern example:
https://gist.github.com/cayter/49d5c256a885d90c399ca6c1eca19f51

We could implement it and hook into `beforeCreate` to validate the row
https://gist.github.com/cayter/49d5c256a885d90c399ca6c1eca19f51#file-repository-ts-L357

But I didn't see anything official. 


I'm a fan of ActiveRecord but Drizzle is super functional. I don't think it is wrong to have a service to create documents with the validation inside. As long as we use it as unique way for creating documents

**Pull request merged**